src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -119,13 +119,14 @@ def _find_inv_for_host(hostname: str, inventory_dict: dict):
         for host in matching_hosts:
             inventory_for_host = _find_inv_for_host(host, self.mgr.cache.devices)
             logger.debug(f"Found inventory for host {inventory_for_host}")
-            drive_selection = DriveSelection(drive_group, inventory_for_host)
+            drive_selection = DriveSelection(drive_group, inventory_for_host,
+                                             self.mgr.cache.get_daemons_with_volatile_status())
```

Comment:
with https://github.com/ceph/ceph/pull/36388 this can be `self.mgr.cache.get_daemons_by_service(drive_group.service_id)`